### PR TITLE
Implemented SecretKey -> Scalar conversion in p256

### PR DESCRIPTION
**Pull request by rozbb**
This is just a small ergonomics issue I ran into when implementing [hpke](https://github.com/rozbb/rust-hpke)

cc @str4d

### CODE REVIEW

File: p256/src/arithmetic/scalar.rs [hunk 1]
```diff
@@ -40,6 +40,17 @@ impl Scalar {
...
 
+    /// Attempts to convert a `SecretKey` (defined in the more generic `elliptic_curve` crate) to a
+    /// `Scalar`
+    ///
+    /// Returns None if the secret's underlying value does not represent a field element.
+    pub fn from_secret(s: SecretKey) -> CtOption<Scalar> {
+        let mut buf = [0u8; 32];
+        buf.copy_from_slice(s.secret_scalar().as_ref());
+        // We can't unwrap() this, since it's not guaranteed that s represents a valid field elem
+        Self::from_bytes(buf)
```
Comment (tarcieri) [inline p256/src/arithmetic/scalar.rs:51]: I believe you can write this as...

```suggestion
        Self::from_bytes(s.secret_scalar().into())
```

...or failing that:

```suggestion
        Self::from_bytes(s.secret_scalar().as_ref().try_into().unwrap())
```

Comment (rozbb) [reply p256/src/arithmetic/scalar.rs:51]: Neither of these worked for me. It doesn't look like `elliptic_curve::Scalar` implements `Into` or `TryInto` for _anything_.

Comment (rozbb) [reply p256/src/arithmetic/scalar.rs]: Fixed!

**Comment by tarcieri** [comment]
Thanks!

**Pull request merged**